src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,8 +33,13 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        splitAddress = address.split(", ");
```

Comment:
- What if the user does not enter all four components

-  Did you run the testcases? Did it pass?